node/bundler.ts

```diff
@@ -186,19 +183,44 @@ const getBasePath = (sourceDirectories: string[], inputBasePath?: string) => {
   return commonPathPrefix(sourceDirectories)
 }
 
-export const addGeneratorFieldIfMissing = (
-  declaration: Declaration,
-  functions: EdgeFunction[],
-  internalFunctionsPath?: string,
-) => {
-  const fullFuncPath = functions?.find((func) => func.name === declaration.function)?.path
+interface MergeWithDeclarationConfigOptions {
+  functionName: string
+  config: FunctionConfig
+  declarations: Declaration[]
+}
 
-  // If function path is in the internalFunctionsPath, we assume it is autogenerated.
-  const isInternal = Boolean(internalFunctionsPath && fullFuncPath && isPathInside(fullFuncPath, internalFunctionsPath))
+// We used to allow the `name` and `generator` fields to be defined at the
+// declaration level. We want these properties to live at the function level
+// in their config object, so we translate that for backwards-compatibility.
+const mergeWithDeclarationConfig = ({ functionName, config, declarations }: MergeWithDeclarationConfigOptions) => {
+  const declaration = declarations?.find((decl) => decl.function === functionName)
 
-  const generatorFallback = isInternal ? 'internalFunc' : undefined
-  return { ...declaration, generator: declaration.generator || generatorFallback }
+  return {
+    ...config,
+    name: declaration?.name || config.name,
+    generator: declaration?.generator || config.generator,
+  }
 }
 
+const addGeneratorFallback = (config: FunctionConfig) => ({
+  ...config,
+  generator: config.generator || 'internalFunc',
+})
+
+interface CreateFunctionConfigOptions {
+  internalFunctionsWithConfig: Record<string, FunctionConfig>
+  declarations: Declaration[]
+}
+
+const createFunctionConfig = ({ internalFunctionsWithConfig, declarations }: CreateFunctionConfigOptions) =>
```

Comment:
nit: I find this function a bit hard to follow. The name isn't super descriptive — it says it creates function configs but it's iterating one of the parameters and calling two other functions inside. I feel like an abstraction like `addGeneratorFallback`, which is simply adding a field, is adding to the confusion more than the benefits it brings.